### DISCUSSION TITLE
Deployment snapshot

### DISCUSSION THREAD

**Discussion by simanadler** [fybrik/fybrik#520] [Developers]
When an M4D deployment is installed it contains components (and specific versions of those components) from different git repositories. Ex: 
* M4D
* Modules such as Flight Server, Data Movement Operator, etc
* DLF (dynamic bucket allocation)
* Connectors to catalog, policy manager, credential manager

Once deployed, there is no way to easily replicate or reinstall.  We need a mechanism to create some kind of snapshot so that the same environment can be re-created.

@roee88 @ronenkat @froesef @shlomitk1 @rohithdv

**Comment by roee88** [comment]
Deployment with specific repo:tag is already possible. For stable releases we have official tags (though we didn't create one yet). For dev builds we have the Custom Build workflow that @froesef wrote. This covers M4D, data movement operator, built-in connectors.

Listing the rest:
- mover image will soon be distributed with tags so it won't just be latest. The tag is specified in the copy module chart. So this should be resolved soon. Related to #500.
- flight module tagging is tracked in https://github.com/IBM/the-mesh-for-data-flight-module/issues/18. @cdoron how would you like to handle it?
- third party modules and connectors are out of our control. The best we can do is document and improve our hello world samples to include tagging.
- dlf is out of our control.

**Reply by simanadler** [reply]
@roee88 While I agree that third party modules are out of our control, I think there should be a way for someone to capture and recreate a deployed environment with ALL of its components.  @froesef In our meeting last week you had an idea how to address this.  Can you please elaborate here?

**Reply by froesef** [reply]
I think we should make sure that everything is installed via helm. That way we can store the used `values.yaml` file for this installation in a private git repository together with the module definitions installed and maybe a little readme about the environment. (e.g. a folder structure like  project/cluster (e.g.  demo1/theshire )) Then at least all info about the deployed environment is in one place and versioned in case we want to upgrade it.

**Reply by simanadler** [reply]
Sounds like a good idea to me @froesef.  @roee88 @ronenkat Your thoughts?  @froesef @ronenkat Would it make sense to create an example for the demos we have been supporting recently?

**Reply by roee88** [reply]
Everything that I wrote still holds, I don't understand what you propose.
